core/src/main/java/org/apache/iceberg/SchemaParser.java

```diff
@@ -248,8 +296,122 @@ public static Schema fromJson(String json) {
       try {
         return fromJson(JsonUtil.mapper().readValue(jsonKey, JsonNode.class));
       } catch (IOException e) {
-        throw new RuntimeIOException(e);
+        throw new RuntimeException(e);
       }
     });
   }
+
+  private static String toJsonString(Map<String, Object> map) {
+    Map<String, String> jsonStringElementsMap = new LinkedHashMap<>();
+    map.entrySet().forEach(
+        entry -> jsonStringElementsMap.put(entry.getKey(), toJsonString(entry.getValue())));
+    return toJsonString(jsonStringElementsMap);
+  }
+
+  private static String toJsonString(List<Object> list) {
+    List<String> jsonStringItemsList = new ArrayList<>();
+    list.forEach(item -> jsonStringItemsList.add(toJsonString(item)));
+    return toJsonString(jsonStringItemsList);
+  }
+
+  private static String toJsonString(Object value) {
+    if (isPrimitiveClass(value)) {
+      return value.toString();
```

Comment:
For `FIXED` and `BINARY`, should do something like `Base64.getEncoder().encodeToString(bytes)`? (also the opposite transformation on the deserialization side).

Reply:
why do we need to treat FIXED and BINARY differently?

Reply:
wrapped into ByteBuffer